OptimizelySDK/Config/DatafileProjectConfig.cs

```diff
@@ -582,6 +650,27 @@ public FeatureFlag GetFeatureFlagFromKey(string featureKey)
             return new FeatureFlag();
         }
 
+        /// <summary>
+        /// Gets the variation associated with an experiment or rollout for a given feature flag key
+        /// </summary>
+        /// <param name="flagKey">feature flag key</param>
+        /// <param name="variationKey">variation key</param>
+        /// <returns></returns>
+        public Variation GetFlagVariationByKey(string flagKey, string variationKey)
+        {
+            if (_FlagVariationMap.TryGetValue(flagKey, out var variations))
+            {
+                foreach (var variation in variations)
+                {
+                    if (variation.Key.Equals(variationKey))
```

Comment:
variation.Key can be `null` which may cause exception, please try variation.Key ==